src/Rules/IRuleInfo.cs

```diff
@@ -44,6 +44,45 @@ public enum RuleIssueType
         Hotspot         // SonarQube serialization = SECURITY_HOTSPOT
     }
 
+    public interface IContext
+    {
+        string Key { get; }
+        string Content { get; }
+    }
+
+    public class Context : IContext
+    {
+        public Context(string key, string content)
+        {
+            Key = key;
+            Content = content;
+        }
+
+        public string Key { get; }
+        public string Content { get; }
+    }
+
+    public interface IDescriptionSection
+    {
+        string Key { get; }
+        string Content { get; }
+        IReadOnlyList<IContext> Context { get; }
+    }
+
+    public class DescriptionSection : IDescriptionSection
+    {
+        public DescriptionSection(string key, string content, IReadOnlyList<IContext> context = null)
+        {
+            Key = key;
+            Content = content;
+            Context = context;
+        }
+
+        public string Key { get; }
+        public string Content { get; }
+        public IReadOnlyList<IContext> Context { get; }
```

Comment:
Have we considered using JetBrains Annotations? [CanBeNull] here would be helpful for understanding

Reply:
I don't think adding a new dependency for this purpose makes sense but I have added some remarks to make it clearer.